src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/IdentityEntityFrameworkBuilderExtensions.cs

```diff
@@ -38,26 +39,28 @@ public static IdentityBuilder AddEntityFrameworkStores<TContext>(this IdentityBu
             where TContext : DbContext
             where TKey : IEquatable<TKey>
         {
-            builder.Services.TryAdd(GetDefaultServices(builder.UserType, builder.RoleType, typeof(TContext), typeof(TKey)));
+            AddStores(builder.Services, builder.UserType, builder.RoleType, typeof(TContext), typeof(TKey));
             return builder;
         }
 
-        private static IServiceCollection GetDefaultServices(Type userType, Type roleType, Type contextType, Type keyType = null)
+        private static void AddStores(IServiceCollection services, Type userType, Type roleType, Type contextType, Type keyType)
         {
-            Type userStoreType;
-            Type roleStoreType;
-            keyType = keyType ?? typeof(string);
-            userStoreType = typeof(UserStore<,,,>).MakeGenericType(userType, roleType, contextType, keyType);
-            roleStoreType = typeof(RoleStore<,,>).MakeGenericType(roleType, contextType, keyType);
-
-            var services = new ServiceCollection();
-            services.AddScoped(
+            var identityUserType = typeof(IdentityUser<>).MakeGenericType(keyType);
+            if (!identityUserType.GetTypeInfo().IsAssignableFrom(userType.GetTypeInfo()))
+            {
+                throw new InvalidOperationException(Resources.NotIdentityUser);
```

Comment:
AFAIR there were a couple of ways to hit the current `TargetInvocationException`. Need to search around to make sure we are covered by this check.